content/200-concepts/100-components/01-prisma-schema/06-relations/420-referential-integrity.mdx

```diff
@@ -0,0 +1,76 @@
+---
+title: 'Referential integrity'
+metaTitle: 'Referential integrity'
+metaDescription: 'Referential integrity lets you define how the consistency of relations is handled.'
+tocDepth: 2
+preview: true
+---
+
+<TopBlock>
+
+Referential integrity means a set of constraints or triggers to handle the consistency of data of a [`@relation`](/reference/api-reference/prisma-schema-reference#relation). It can mean preventing a deletion of a record referred from other records, or it can be a cascading action defined in the field's [`referential actions`](/concepts/components/prisma-schema/relations/referential-actions). 
+
+The referential integrity can either be handled in the database using foreign keys, or in the Prisma Client. There are pros and cons on both. In general, if the database supports foreign keys, it is usually the preferred choice and the default for SQL databases when using Prisma.
+
+In certain cases, either when the database doesn't support foreign keys, or when they are not preferred due to scaling reasons, Prisma can disable them and handle the the referential integrity in the client.
+
+</TopBlock>
+
+## Setting the integrity in the datasource
+
+Changing the referential integrity can be done in the [`datasource`](/reference/api-reference/prisma-schema-reference#datasource) block, using the `referentialIntegrity` parameter. The default for all SQL databases is `foreignKeys` and MongoDB defaults to `prisma`, which is the only setting that can be used with the connector.
+
```

Comment:
```suggestion
Changing the referential integrity can be done in the [`datasource`](/reference/api-reference/prisma-schema-reference#datasource) block, using the `referentialIntegrity` parameter. The default for all SQL databases is `foreignKeys` and MongoDB defaults to `prisma`, which is the only setting that can be used with the connector.

Because referential integrity is currently a preview feature, you need to explicitly define that in the generator block with `previewFeatures = ["referentialIntegrity"]`.
```